src/components/buttons/ha-progress-button.ts

```diff
@@ -0,0 +1,115 @@
+import "@material/mwc-button";
+import {
+  css,
+  CSSResult,
+  customElement,
+  html,
+  LitElement,
+  property,
+  PropertyValues,
+  query,
+  TemplateResult,
+} from "lit-element";
+
+import "../ha-circular-progress";
+
+export interface HaProgressButtonElement extends HTMLElement {
+  disabled: boolean;
+  progress: boolean;
+}
+
+@customElement("ha-progress-button")
+class HaProgressButton extends LitElement {
+  @property({ type: Boolean }) public disabled = false;
+
+  @property({ type: Boolean }) public progress = false;
+
+  @query("#container") private _buttonContainer!: HTMLDivElement;
+
+  public render(): TemplateResult | void {
+    return html`
+      <div id="container">
```

Comment:
Do we need this container? Should be able to set attributes on the element itself